### PR TITLE
Add JSON Parse Functionality to Servlet (Walkthrough-3)

### PR DESCRIPTION
**Pull request by samuelsogbesan**
This branch was branched off of servlet-2 and relies on pending pull requests in its early commit history.

- **Add hard-coded Arraylist to DataServlet and convert it to JSON as the response for "/data"**
- **Perform a fetch on window load from script.js**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,24 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
```
Comment (krasimirgg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: [put imported names in ASCII order](https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -19,14 +19,24 @@
...
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> json = new ArrayList<String>();
+  
+  @Override
+  public void init(){
+    json.add("Surprise!");
```
Comment (krasimirgg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Filling the `json` array in the init() method right now feels a bit of an overkill.
Consider instead assigning directly to the `json` member variable on line 28 above using a helper like [`Arrays.asList`](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#asList-T...-).

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -10,4 +10,23 @@
...
+function printMessageFromServer(){
+  fetch("/data")
+  .then(data => data.text())
+  .then(message => {
+    var container = document.getElementById("data-container");
+    container.innerText = message;
+  })
+}
+
+//window.onload = () => document.getElementById("data-button").addEventListener("click",printMessageFromServer);
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: remove this commented out code

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -10,4 +10,23 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+function printMessageFromServer(){
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: update this similarly to [your other pull request](https://github.com/samuelsogbesan/step/pull/17/files)

File: portfolio/src/main/webapp/style.css [hunk 5]
```diff
@@ -11,7 +11,7 @@ body {
 }
 
 #fullpage {
-  min-width: 100vw;
+min-width: 100vw;
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/style.css]: fix indentation and sort order

**Review by krasimirgg — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (samuelsogbesan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Definition changed in commits b36b89e and 3bb0c43

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Removed with commit 56036cd

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Added space between () and { with commit 418b853

File: portfolio/src/main/webapp/style.css (hunk 5 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/style.css]: Updated with commit e240e53

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (samuelsogbesan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Updated with commit b849b6c

Comment (samuelsogbesan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Updated again with commit a0c479e

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -14,19 +14,25 @@
...
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private List<String> json = Arrays.asList(new String[]{"Surprise!","Boo!","Woah!"});
```
Comment (krasimirgg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: [add a space after `,` in array literals](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace) (rule 5)

Or even more concise:
`private List<String> json = Arrays.asList("Surprise!", "Boo!", "Woah!");`
Explanation: You don't need to explicitly construct a `String[]` as a parameter.
In Java, when an argument is declared with `...` after the type, as in [asList(T... a)](https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#asList(T...) ), you can just call it with an arbitrary number of arguments; the array creation is implicit.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -10,4 +10,20 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+const add = (datapoint) => {
+  const container = document.getElementById("data-container");
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js:16]: [use single quotes](https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes), here and elsewhere
(it's confusing to have a different advice for Java and Javascript, but yolo)

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -10,4 +10,20 @@
...
+}
+
+const printJSONFromServer = () => {
+  container = document.getElementById("data-container");
+  fetch('/data')
+  .then(data => data.json())
+  .then(object => object.forEach(datapoint => add(datapoint)))
+}
+
+window.onload = () => document.getElementById("data-button").addEventListener("click",printJSONFromServer);
```
Comment (krasimirgg) [inline portfolio/src/main/webapp/script.js]: add a space after `,`

**Review by krasimirgg — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js:16]: Updated with commit 64d662a

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (samuelsogbesan) [reply portfolio/src/main/webapp/script.js]: Updated in commit b9da75c

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (samuelsogbesan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Thanks for the tip! Changes made through commits 0a1f683 and 58a6bec

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (krasimirgg) [reply portfolio/src/main/webapp/script.js:16]: `"data-container"` on line 16 is the last instance of double quotes remaining.

**Review by krasimirgg — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (krasimirgg) [reply portfolio/src/main/webapp/script.js:16]: "data-container" on line 16 is the last instance of double quotes remaining.

**Review by krasimirgg — CHANGES_REQUESTED** [review]

**Comment by samuelsogbesan** [comment]
This functionality has been re-implemented in PRs #28 #29 and #30 directly for the comment section. I'll close this pull request as to not cause confusion.

**Pull request closed without merging**